staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy_test.go

```diff
@@ -134,5 +134,9 @@ func TestPrepareForUpdate(t *testing.T) {
 		if !reflect.DeepEqual(tc.obj, tc.expected) {
 			t.Errorf("test %d failed: expected: %v, got %v", index, tc.expected, tc.obj)
 		}
+		errorList := strategy.Validate(context.TODO(), tc.obj)
+		if !reflect.DeepEqual(len(errorList), 0) {
+			t.Errorf("default update validation fails")
+		}
```

Comment:
does this test exercise all of the following cases on a schema with a set and map list:
1. new object with valid set/map lists (should pass validation)
2. new object with invalid set/map lists where the old object was valid (should fail validation)
3. new object with invalid set/map lists where the old object was also invalid (should pass validation)

test case 2 should pass on this PR and fail on master